AutoCollection/StatsBeat.ts

```diff
@@ -0,0 +1,218 @@
+import os = require("os");
+import TelemetryClient = require("../Library/TelemetryClient");
+import Constants = require("../Declarations/Constants");
+import AzureVirtualMachine = require("../Library/AzureVirtualMachine");
+import Config = require("../Library/Config");
+import Context = require("../Library/Context");
+import { MetricTelemetry } from "../Declarations/Contracts";
+
+const STATS_BEAT_LANGUAGE = "node";
+const STATS_BEAT_IKEY = "";
+
+
+export class StatsBeat {
+
+    // Counters
+    private _lastRequests: { totalRequestCount: number; time: number; };
+    private _totalRequestCount: number = 0;
+    private _totalSuccesfulRequestCount: number = 0;
+    private _totalFailedRequestCount: number = 0;
+    private _retryCount: number = 0;
+    private _exceptionCount: number = 0;
```

Comment:
how does node.js handle multiple threading? these availables are needed to be thread-safe?

Reply:
No need to worry about multiple threads in Javascript, this values must be good and counts will be associated by specific Telemetry Clients